src/dune_sexp/decoder.mli

```diff
@@ -120,8 +120,9 @@ val kind : (kind, _) parser
     end of sequence is reached. *)
 val repeat : 'a t -> 'a list t
 
-(** Like [repeat] but the list of elements must be non-empty. *)
-val repeat1 : 'a t -> 'a list t
+(** Like [repeat] but the list of elements must be non-empty. [check_multi] is
+    performed when the parsed list has length > 1. *)
+val repeat1 : ?check_multi:unit t -> 'a t -> 'a list t
```

Comment:
why do we need to pass check_multi as an argument. Can't we always rewrite `repeat1 ~check_multi t` as:

```
let* x = repeat1 y in
let+ () = check_multi x in
x
```

Reply:
the reason I pushed it into `repeat1` is that this avoids the need for `let*` in callers.
I see 2 benefits in this:
- these call sites stay simpler
- this leaves the door open to making `Decoder` purely applicative (`repeat1` would be a constructor)

Reply:
The call sites are a little shorter, but not simpler. One has to understand this new argument, and keep in mind when it's requireed. The trivial construct it replaces is much simpler than all that.

I'm not sure how much of a goal it is it to make the parser applicative, but it wouldn't be right to do it by adding new arguments to capture every behavior.

The right way to go would be to add a combinator that does what you want. Perhaps something like:

```
val ( <* ) : 'a t -> unit t -> 'a t
```

Reply:
Allright, let's scrap that idea.